front/src/applications/operationalStudies/views/Scenario/components/MacroEditor/utils.ts

```diff
@@ -184,6 +184,33 @@ export const getSavedMacroNodes = async (
   return result;
 };
 
+/**
+ * Check if every new nodes (from import) are already in the DB (by path_item_key).
+ * If not we persist them.
+ */
+export const persistNodesFromFileImport = async (
+  state: MacroEditorState,
+  dispatch: AppDispatch
+) => {
+  const saved = await getSavedMacroNodes(state, dispatch);
```

Comment:
We already have `savedNodes` loaded in `loadAndIndexNge()`. Instead of loading them again from the backend, can we pass them as function arguments?